src/util/tr.js

```diff
@@ -496,75 +499,130 @@ const _tr = (msg, {
+    }
 
-    iframe.setAttribute('style', 'z-index: 2147483647; display:block; width: 100% !important; border:0px !important; border-color:none !important;')
     msgDiv.appendChild(iframe)
-    const ifrm = (iframe.contentWindow) ? iframe.contentWindow : (iframe.contentDocument.document) ? iframe.contentDocument.document : iframe.contentDocument
-    const doc = ifrm.document
 
     // Dispatch event for popup box/banner close
     const closeCampaign = new Event('CT_campaign_rendered')
     document.dispatchEvent(closeCampaign)
 
-    doc.open()
-    doc.write(html)
-
     if (displayObj['custom-editor']) {
-      appendScriptForCustomEvent(targetingMsgJson, doc)
+      html = appendScriptForCustomEvent(targetingMsgJson, html)
+    }
+    iframe.srcdoc = html
+
+    // const ua = navigator.userAgent.toLowerCase()
+    let contentDiv
+
+    const ActionType = {
+      ADJUST_IFRAME_HEIGHT: 'adjustIFrameHeight',
+      UPDATE_HEIGHT: 'update height',
+      GET_NOTIFICATION: 'getnotif',
+      EVENT: 'Event',
+      PROFILE: 'Profile',
+      OUL: 'OUL',
+      CLOSE_BOX_POPUP: 'closeBoxPopUp',
+      CLOSE_BANNER_POPUP: 'closeBannerPopUp'
     }
-    doc.close()
 
-    const adjustIFrameHeight = () => {
-      // adjust iframe and body height of html inside correctly
-      contentHeight = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv').scrollHeight
-      if (displayObj['custom-editor'] !== true && !isBanner) {
-        contentHeight += 25
+    const handleMessage = (event, displayObj, divId) => {
+      let heightAdjust, boxWrapper, bannerWrapper // Declare variables outside of switch
+
+      switch (event?.data?.action) {
+        case ActionType.UPDATE_HEIGHT + displayObj.layout:
+          heightAdjust = document.getElementById(divId)
+          if (heightAdjust) {
+            heightAdjust.style.margin = '0px'
+            heightAdjust.style.height = `${event.data.value}px`
+          }
+          break
+        case ActionType.GET_NOTIFICATION + displayObj.layout:
+          window.clevertap.renderNotificationClicked(event.data.value)
+          break
+        case ActionType.EVENT:
+          window.clevertap.event.push(event.data.value)
+          break
+        case ActionType.PROFILE:
+          window.clevertap.profile.push(event.data.value)
+          break
+        case ActionType.OUL:
+          window.clevertap.onUserLogin.push(event.data.value)
+          break
+        case ActionType.CLOSE_BOX_POPUP:
+          setTimeout(() => {
+            boxWrapper = window.document.getElementById('wizParDiv0')
+            boxWrapper && boxWrapper.remove()
+          }, 0)
+          break
+        case ActionType.CLOSE_BANNER_POPUP:
+          setTimeout(() => {
+            bannerWrapper = window.document.getElementById('wizParDiv2')
+            bannerWrapper && bannerWrapper.remove()
+          }, 0)
+          break
+        default:
+          // Handle unknown action
```

Comment:
lets have a log here. We need to know whether someone is passing some unknown action